### ISSUE TITLE
Remove quoted-antipattern workaround (hook exclusions, regression test, doc caveat) once H2/H4/H5 scope-classifier fix lands

### ISSUE THREAD

**Issue by roli-lpci** [roli-lpci/lintlang#32]
**Severity:** blocker

**Reproduction:**

1. The shared pre-commit hook (`.git/hooks/pre-commit`, not per-worktree) excludes two files from lintlang's own dogfooding scan, with the reason inlined as a comment:

```bash
# Exclude this repo's own documentation. These files DESCRIBE antipatterns and
# quote them as examples ("keep trying until", "be concise"), and the detectors
# read quoted examples as live instructions. That is a real false-positive class
# tracked separately; it is not a reason to lint a reference manual as if it
# were an agent config.
AGENT_CONFIGS=$(echo "$AGENT_CONFIGS" | grep -vE '^(llms-full\.txt|tests/test_quoted_example_regression\.py)$' || true)
```

2. Confirmed this exclusion is currently load-bearing — without it, scanning `llms-full.txt` fails outright:

```
$ cd ~/github-projects/lintlang/.claude/worktrees/scope-classifier-0.3.9
$ ../../../.venv/bin/python -m lintlang scan llms-full.txt --fail-on fail
...
Verdict: FAIL — llms-full.txt has CRITICAL/HIGH findings
  ❌ FAIL — 4 CRITICAL, 2 HIGH, 1 MEDIUM, 9 LOW

  H2: Missing Constraint Scaffolding
    !! [CRITICAL] system_prompt
      Unbounded retry loop — 'keep trying until' needs an explicit limit.
      Evidence: "unded retry loops ("keep trying until")` is reported as containing an unbound"
    !! [CRITICAL] system_prompt
      Negative termination condition — rephrase as a positive bound.
      Evidence: "ative termination ("don't stop until"), "continue until" without limits, mis"
    !! [CRITICAL] system_prompt
      Unbounded continuation — add an explicit termination condition.
    ... (plus H4 HIGH findings on "remember everything", H5 LOW findings, H6 MEDIUM)

$ echo "exit: $?"
exit: 1
```

Every CRITICAL/HIGH finding is text lintlang's own manual *quotes* to describe the antipattern it warns about (e.g. `("keep trying until")` inside a sentence explaining what H2 catches), not a live instruction. This is the exact bug the companion issue targets.

3. Read `tests/test_quoted_example_regression.py` in full and ran it — both tests currently pass, confirming their documented shape:

```
$ ../../../.venv/bin/python -m pytest tests/test_quoted_example_regression.py -v
tests/test_quoted_example_regression.py::test_quoted_antipatterns_currently_fire PASSED
tests/test_quoted_example_regression.py::test_the_same_text_as_a_real_instruction_must_always_fire PASSED
```

- `test_quoted_antipatterns_currently_fire` runs `detect_h2` + `detect_h5` against a documentation fixture (`## H2: Missing Constraint Scaffolding\nCatches: unbounded retry loops ("keep trying until")...`) and asserts `findings` is truthy — i.e. it asserts the bug's presence. Its own docstring says: "Fails when the scope classifier lands... If this test fails, the fix has arrived. Update this module to assert the corrected behaviour and remove the `llms-full.txt` exclusion from `.git/hooks/pre-commit`."
- `test_the_same_text_as_a_real_instruction_must_always_fire` runs `detect_h2` against a genuine live directive ("Keep trying until the ticket is resolved.\nDo not stop until the customer confirms.") and asserts findings are non-empty. This is the permanent guard rail and must never be weakened.
- The module docstring also states this test file is itself excluded from the repo's own pre-commit lint "for the same reason `llms-full.txt` is," and says explicitly: "Both exclusions live in `.git/hooks/pre-commit` and should be deleted when the classifier lands."

4. `llms-full.txt` line 122 has a section titled `### Known false positive: quoted antipatterns` (verified via `grep -n "Known false positive: quoted antipatterns" llms-full.txt`) documenting this exact limitation to users:

> H2, H4 and H5 do not distinguish reported speech from a directive, so documentation that quotes an antipattern in order to warn against it is read as issuing it. A style guide containing `unbounded retry loops ("keep trying until")` is reported as containing an unbounded retry loop. Point the tool at agent configs rather than at prose about agent configs, or use `.lintlangignore`. This project excludes its own reference manual from its own pre-commit hook for exactly this reason.

**Expected:** once the companion fix lands (wiring `src/lintlang/preflight/scope.py`'s `analyze_scope()` DIRECT/QUOTED/CODE/HYPOTHETICAL/NEGATED classifier into `detect_h2`/`detect_h4`/`detect_h5`), "clean" means all of:

- Both exclusion lines gone from `.git/hooks/pre-commit`: the inline comment block explaining the false-positive workaround, and the `grep -vE '^(llms-full\.txt|tests/test_quoted_example_regression\.py)$'` filter line — the hook goes back to scanning every staged `.txt`/`.py` agent-config-pattern file with no special-casing.
- `lintlang scan llms-full.txt --fail-on fail` exits 0 (or at minimum produces no CRITICAL/HIGH findings sourced from quoted antipattern text) — i.e. a real `git commit` touching `llms-full.txt` succeeds without needing the hook exclusion.
- `test_quoted_antipatterns_currently_fire` converted (not deleted) to assert the corrected behavior: no findings (or only the correct/expected finding set) on `DOCUMENTATION_DESCRIBING_ANTIPATTERNS`. Rename it to reflect the fixed behavior (e.g. `test_quoted_antipatterns_no_longer_fire`) and update/remove the module docstring's now-stale incident narrative as appropriate.
- `test_the_same_text_as_a_real_instruction_must_always_fire` left untouched and still passing — this is the permanent regression guard and must not be weakened by the classifier wiring.
- The `### Known false positive: quoted antipatterns` section removed from `llms-full.txt` (currently at line 122, one paragraph, immediately before the `### H2: Missing Constraint Scaffolding` section), since it would otherwise document a bug that no longer exists.

**Scope:** files that must change: `.git/hooks/pre-commit`, `tests/test_quoted_example_regression.py`, `llms-full.txt`.

This issue is **BLOCKED BY** the companion issue (filed in this same audit round) that fixes the root defect — H2, H4 and H5 in `src/lintlang/patterns.py` reading quoted/reported documentation text as live instructions, via wiring in the existing `analyze_scope()` classifier from `src/lintlang/preflight/scope.py`. Do not implement this cleanup before that companion fix lands: removing the `.git/hooks/pre-commit` exclusion before the underlying detector fix exists would immediately break this repo's own pre-commit hook on any real commit touching `llms-full.txt` or the regression test file, since the false positive demonstrated above is still live.

**Done when:**

```bash
# 1. Converted test passes
pytest tests/test_quoted_example_regression.py -v
# both tests PASS, including the converted test_quoted_antipatterns_currently_fire

# 2. Both exclusion patterns are gone from the hook
grep -E 'llms-full\.txt|tests/test_quoted_example_regression\.py' .git/hooks/pre-commit
# returns nothing (exit 1 / no match)

# 3. A real commit touching both files succeeds without the hook blocking it
git add llms-full.txt tests/test_quoted_example_regression.py
git commit -m "..."
# hook runs lintlang scan on both files as ordinary agent configs and does not block the commit
```

**Comment by roli-lpci** [comment]
Implemented on `feat/scope-classifier-0.3.9` (worktree `.claude/worktrees/scope-classifier-0.3.9`), on top of `219de2e` (companion scope-classifier fix, #33, already landed on this branch).

**Done, verified:**
- `.git/hooks/pre-commit`: removed the comment block and the `grep -vE '^(llms-full\.txt|tests/test_quoted_example_regression\.py)$'` line. `grep -E 'llms-full\.txt|tests/test_quoted_example_regression\.py' .git/hooks/pre-commit` now returns nothing (exit 1), matching the issue's Done-when #2.
- `llms-full.txt`: removed the `### Known false positive: quoted antipatterns` section (lines 122-125). `lintlang scan llms-full.txt --fail-on fail` now exits 0 (1 MEDIUM finding, no CRITICAL/HIGH):
  ```
  $ PYTHONPATH=src ~/github-projects/lintlang/.venv/bin/python -m lintlang scan llms-full.txt --fail-on fail
  ⚠️ REVIEW — 1 MEDIUM
  exit: 0
  ```
- `tests/test_quoted_example_regression.py`: converted `test_quoted_antipatterns_currently_fire` → `test_quoted_antipatterns_no_longer_fire`, asserting `not findings` instead of `findings`. Updated the module docstring to describe the fixed state. Confirmed pre-conversion the old assertion was already failing (`assert []` truthy-check failed, findings now empty) — that was the starting point per the implementer rules. Post-conversion, both tests in the file pass. `test_the_same_text_as_a_real_instruction_must_always_fire` is byte-identical to before (untouched).
- Full regression: `pytest -q` → 286 passed, 76 subtests passed. `ruff check src/ tests/` → all checks passed.

**Not done — genuine blocker found, filed as #40:**

Done-when #3 (`git add llms-full.txt tests/test_quoted_example_regression.py && git commit` succeeds without the hook blocking) does **not** hold, even with the scope-classifier fix correctly wired in and both exclusions removed. `llms-full.txt` alone scans clean, but the hook also scans `tests/test_quoted_example_regression.py` itself (`.py` is in `AGENT_CONFIG_PATTERNS`, and `parsers.py` has no `.py`-specific handling, so it treats the whole raw file as one `system_prompt`). The permanent guard-rail fixture in `test_the_same_text_as_a_real_instruction_must_always_fire` — "Keep trying until the ticket is resolved. Do not stop until the customer confirms." — is *designed* to be indistinguishable from a live directive (that's the point of the guard rail), so `detect_h2` correctly fires CRITICAL on it, and a real `git commit` of both files is aborted by the hook (exit 1, verified for real, not just read):

```
$ git add llms-full.txt tests/test_quoted_example_regression.py
$ PYTHONPATH=src git commit -m "..."
❌ lintlang violations found in: tests/test_quoted_example_regression.py
  !! [CRITICAL] tests/test_quoted_example_regression.py:68 > system_prompt
    Unbounded retry loop — 'keep trying until' needs an explicit limit.
COMMIT EXIT: 1
```

This is not the quoted-antipattern bug (that part is fixed and confirmed clean) — it's a second, independent, structural reason the old exclusion existed, that #32 didn't anticipate: `test_the_same_text_as_a_real_instruction_must_always_fire`'s fixture will *always* trip this hook's naive whole-file scan, correctly, by design, forever, regardless of any detector fix. Fixing it requires either changing what `.py` files the hook scans or how `.py` files get parsed — both out of #32's stated scope (`.git/hooks/pre-commit`, `tests/test_quoted_example_regression.py`, `llms-full.txt` only) and both would mean re-adding some form of special-casing, which #32 explicitly asked to eliminate, or touching the guard-rail fixture itself, which #32 explicitly forbade.

Filed #40 with full reproduction and a scoped "Done when" for the actual fix. Per the Agent PR Protocol implementer rules ("If the issue's `Done when:` can't be expressed as a check [as originally stated], stop and say so on the issue rather than guessing" / "related defects noticed along the way get filed as new issues, not folded into this diff"), I did not force a commit through `--no-verify` (forbidden without explicit request) and did not fix #40's root cause here (out of #32's scope).

**State left for review:** the three files are edited and staged in the worktree, uncommitted. No commit exists yet — creating one that closes #32 as originally specified isn't possible until #40 is resolved (or #32's Done-when #3 is explicitly narrowed to "a commit touching `llms-full.txt` alone").
